src/rgb_led.rs

```diff
@@ -76,3 +78,58 @@ impl<F: FullDuplex<u8>> LedStrip<F> {
         }
     }
 }
+
+/// U64Instant::elapsed() tries to correct the u32 overflow of the underlying Instant. It is
+/// supposed to be accurate as long as the function is called frequently enough i.e. at least
+/// once per 1 minute 29 seconds.
+struct U64Instant {
+    elapsed: u64,
+    last_elapsed: u32,
+    instant: Instant,
+}
+
+impl From<Instant> for U64Instant {
+    fn from(instant: Instant) -> Self {
+        let elapsed = instant.elapsed();
+
+        Self {
+            elapsed: elapsed as u64,
+            last_elapsed: elapsed,
+            instant, 
+        }
+    }
+}
+
+impl U64Instant {
+    fn elapsed(&mut self) -> u64 {
+        let mut diff = self.instant.elapsed() as i32 - self.last_elapsed as i32;
+        if diff < 0 {
+            diff += u32::MAX as i32 + 1;
```

Comment:
`u32::MAX as i32` doesn't that always evaluate to `-1`?

Reply:
oh yeah it does...